src/wrapper/java_vm/vm.rs

```diff
@@ -11,7 +11,16 @@ use log::{debug, error};
 use crate::{errors::*, sys, JNIEnv};
 
 #[cfg(feature = "invocation")]
-use crate::InitArgs;
+use {crate::InitArgs, cfg_if::cfg_if, std::os::raw::c_void};
+
+#[cfg(feature = "invocation-dyn")]
+use {
+    once_cell::sync::OnceCell,
+    std::{ffi::OsStr, path::PathBuf},
+};
+
+#[cfg(feature = "invocation-dyn")]
+static LIBJVM: OnceCell<(String, libloading::Library)> = OnceCell::new();
```

Comment:
It's been too long; I don't remember exactly why I did that. You're right that the OS does its own caching and won't load the same library twice, so it should be fine to remove that.

I do know that, [per the JNI specification, creation of multiple JVMs in a single process is not supported.](https://docs.oracle.com/en/java/javase/11/docs/specs/jni/invocation.html#jni_createjavavm) `JNI_CreateJavaVM` will return an error if called when a JVM is already running in the same process. I don't know what will happen if multiple JVM *libraries* are loaded into the same process, but I'm guessing they'll successfully misbehave instead of returning an error (i.e. undefined-ish behavior). Among other things, the JVM installs signal handlers for a wide variety of signals including `SIGBUS` and `SIGSEGV`, so with multiple JVMs running in the same process, only the last one created would receive any signals, and probably abort the process upon receiving a signal it wasn't expecting.

Still, that doesn't mean we should stop people from loading multiple JVM libraries. It is possible, and probably safe, to `DestroyJavaVM`, load a different JVM library, and `JNI_CreateJavaVM` with that one.

Reply:
Thanks for pointing at the spec re: non-support for loading JVMs - that makes sense.

Yeah I was more thinking along the lines of an app (or maybe a test framework) that might try and load one JVM and then tear it down before loading a different one (as opposed to trying your luck running multiple in parallel).

It's not something I'm aware of anyone doing I was more just surprised to see the global static and it just seemed like it could (potentially) interfere with that kind of use case and it also wasn't obvious what disadvantage there would be from removing it anyway as a small simplification.

Thanks for the response